src/aiida_quantumespresso/parsers/parse_raw/ph.py

```diff
@@ -190,35 +231,34 @@ def detect_important_message(logs, line):
                 raise QEOutputParsingError('Unable to convert wall_time in seconds.')
             break
 
-    # Parse number of q-points and number of atoms
+    parsed_data['num_q_found'] = 0
+
     for count, line in enumerate(lines):
 
         detect_important_message(logs, line)
 
         if 'q-points for this run' in line:
             try:
-                num_qpoints = int(line.split('/')[1].split('q-points')[0])
-                if (
-                    'number_of_qpoints' in list(parsed_data.keys()) and num_qpoints != parsed_data['number_of_qpoints']
-                ):
-                    logs.warning.append('Number q-points found several times with different values')
-                else:
-                    parsed_data['number_of_qpoints'] = num_qpoints
-            except Exception:
-                logs.warning.append('Error while parsing number of q points.')
+                parsed_data['number_of_qpoints'] = int(line.split('/')[1].split('q-points')[0])
+                parsed_data['q_points'] = parse_qpoints(lines[count + 2:count + parsed_data['number_of_qpoints'] + 2])
+
+            except Exception as exc:
+                logs.warning.append(f'Error while parsing number of q points: {exc}')
 
         elif 'q-points)' in line:
             # case of a 'only_wfc' calculation
             try:
-                num_qpoints = int(line.split('q-points')[0].split('(')[1])
-                if (
-                    'number_of_qpoints' in list(parsed_data.keys()) and num_qpoints != parsed_data['number_of_qpoints']
-                ):
-                    logs.warning.append('Number q-points found several times with different values')
-                else:
-                    parsed_data['number_of_qpoints'] = num_qpoints
-            except Exception:
-                logs.warning.append('Error while parsing number of q points.')
+                parsed_data['number_of_qpoints'] = int(line.split('q-points')[0].split('(')[1])
+                parsed_data['q_points'] = parse_qpoints(lines[count + 2:count + parsed_data['number_of_qpoints'] + 2])
+
+            except Exception as exc:
+                logs.warning.append(f'Error while parsing number of q points: {exc}')
+
+        # In case a single q-point is provided at the end of the input file, there is no summary of the q-points at the
+        # start of the stdout. Then we have to parse the q-point further down the output file.
+        elif 'number_of_qpoints' not in parsed_data and 'Calculation of q =' in line:
+            parsed_data['number_of_qpoints'] = 1
+            parsed_data['q_points'] = {0: [float(coord) for coord in line.split('=')[-1].split()]}
 
```

Comment:
Need to test with an actual run - but I think the above should fix the following comment

> When a PhBaseCalculation for 1 q-point has a PhCalculation that hits the wall time and then properly restarts, the final mode symmetries are not parsed. This is probably due to some logic in parses/parse_raw/ph.py - will try look into it shortly

Reply:
seems good to go

Reply:
Just amazing. So QE _doesn't_ print the `1 q-points for this run` for the first run of a single q-point, but it _does_ for a restart run. And why not just call the file `dynamical-matrix-1`? Too consistent?

I'm going to ask @giovannipizzi if it's possible to organise a Trieste visit and just spend as much time as is needed to move all the outputs to XML. At this point I think it's more efficient 😂.

Reply:
I propose an alternative fix in [47f5cc7](https://github.com/aiidateam/aiida-quantumespresso/pull/818/commits/47f5cc7c92f491ea5bf5afa19b4599927de669d9). It's not super clean, but I'm loathe to start fidgeting with the dynamical matrix parsing too much since we don't have many tests for other use cases of `ph.x`. This change has the added benefit that the output is more consistent, i.e. now for both single and multiple kpoint(s) the first output key is `dynamical_matrix_1`.